src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalescingAssignmentTests.cs

```diff
@@ -0,0 +1,1239 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.NullCoalescingAssignment)]
+    public class NullCoalescingAssignmentTests : CompilingTestBase
+    {
+        [Fact]
+        public void LocalLvalue()
+        {
+
+            CompileAndVerify(@"
+using System;
+public class C
+{
+    public static void Main()
+    {
+        int? i1 = null;
+        Console.WriteLine(i1 ??= 0);
+    }
+}
+", expectedOutput: "0").VerifyIL("C.Main()", @"
+{
+  // Code size       39 (0x27)
+  .maxstack  1
+  .locals init (int? V_0)
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  initobj    ""int?""
+  IL_0008:  ldloc.0
+  IL_0009:  stloc.0
+  IL_000a:  ldloca.s   V_0
+  IL_000c:  call       ""bool int?.HasValue.get""
+  IL_0011:  brtrue.s   IL_001b
+  IL_0013:  ldc.i4.0
+  IL_0014:  newobj     ""int?..ctor(int)""
+  IL_0019:  br.s       IL_001c
+  IL_001b:  ldloc.0
+  IL_001c:  box        ""int?""
+  IL_0021:  call       ""void System.Console.WriteLine(object)""
+  IL_0026:  ret
+}
+");
+        }
+
+        [Fact]
+        public void FieldLvalue()
+        {
+
+            CompileAndVerify(@"
+public class C
+{
+    int? f1 = null;
+
+    public static void Main()
+    {
+        var c = new C();
+        Test(c.f1 ??= 0);
+    }
+    public static void Test(int? test) {}
+}
+").VerifyIL("C.Main()", @"
+{
+  // Code size       46 (0x2e)
+  .maxstack  3
+  .locals init (C V_0,
+                int? V_1,
+                int? V_2)
+  IL_0000:  newobj     ""C..ctor()""
+  IL_0005:  stloc.0
+  IL_0006:  ldloc.0
+  IL_0007:  ldfld      ""int? C.f1""
+  IL_000c:  stloc.1
+  IL_000d:  ldloca.s   V_1
+  IL_000f:  call       ""bool int?.HasValue.get""
+  IL_0014:  brtrue.s   IL_0027
+  IL_0016:  ldloc.0
+  IL_0017:  ldc.i4.0
+  IL_0018:  newobj     ""int?..ctor(int)""
+  IL_001d:  dup
+  IL_001e:  stloc.2
+  IL_001f:  stfld      ""int? C.f1""
+  IL_0024:  ldloc.2
+  IL_0025:  br.s       IL_0028
+  IL_0027:  ldloc.1
+  IL_0028:  call       ""void C.Test(int?)""
+  IL_002d:  ret
+}
+");
+        }
+
+        [Fact]
+        public void IndexerLvalue()
+        {
+
+            CompileAndVerify(@"
+using System;
+public class C
+{
+    int? f1 = null;
+
+    public static void Main()
+    {
+        int?[] a = new int?[1];
+        Console.WriteLine(a[0] ??= 1);
```

Comment:
>0 [](start = 28, length = 1)

Please use an expression with side-effects as the index.

Reply:
Please verify the indexed element is modified.

---
In reply to: [199829846](https://github.com/dotnet/roslyn/pull/28253#discussion_r199829846) [](ancestors = 199829846)